packages/webview_flutter/webview_flutter/lib/platform_interface.dart

```diff
@@ -173,15 +174,25 @@ abstract class WebViewPlatformController {
 
   /// Loads the specified URL.
   ///
-  /// If `headers` is not null and the URL is an HTTP URL, the key value paris in `headers` will
-  /// be added as key value pairs of HTTP headers for the request.
+  /// If [WebViewRequest] object is not null and the URL is a HTTP URL
+  /// then the following rules apply:
+  ///
+  /// - [WebViewRequest.method] must be one of the supported HTTP methods
+  /// in the [WebViewLoadMethod].
+  ///
+  /// - If [WebViewRequest.headers] is not null, the key value pairs in
+  /// [WebViewRequest.headers] will be added as key value pairs of HTTP headers
+  /// for the request.
+  ///
+  /// - If [WebViewRequest.body] is not null, it will be added as HTTP body
+  /// for the request.
   ///
   /// `url` must not be null.
   ///
   /// Throws an ArgumentError if `url` is not a valid URL string.
   Future<void> loadUrl(
     String url,
-    Map<String, String>? headers,
+    WebViewRequest? request,
```

Comment:
I don't think this should be nullable; a request must have a method. (If URL gets folded in, it will definitely not be nullable).